ArduCopter/AP_Arming.cpp

```diff
@@ -596,6 +596,21 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
         return false;
     }
 
+    // Do not arm if EKF lane 1 is not used as primary during arming, when using EK2.
+    const NavEKF2 &ekf2 = AP::ahrs_navekf().get_NavEKF2_const();
+    if (ekf2.currentPrimary() != 0) {
+        check_failed(true, "EKF lane 1 not in use as primary");
+        return false;
+    }
+
+    // Do not arm if EKF lane 1 is not used as primary during arming, when using EK3
```

Comment:
What is the outcome of `ekf3.currentPrimary()` when not using EK3? Should there be a check for which EKX we're using?

Reply:
Additionally, is there only one primary? if so, can the second `if` be an `else if`?

Reply:
As per the new changes of adding a check with EKF_TYPE, else if makes more sense. Done!